iModelCore/ecobjects/src/ECInstance.cpp

```diff
@@ -2687,14 +2687,22 @@ struct  InstanceXmlReader
                 return InstanceReadStatus::ECSchemaNotFound;
                 }
 
+            // do not search for the class if its schema is going to be pruned.
+            SchemaKey key;
+            ECObjectsStatus schemaNameParseStatus;
+            if (ECObjectsStatus::Success == (schemaNameParseStatus = SchemaKey::ParseSchemaFullName(key, m_fullSchemaName.c_str())) && m_context.IsSchemaToBePruned(key))

```

Comment:
```suggestion
            ECObjectsStatus schemaNameParseStatus = SchemaKey::ParseSchemaFullName(key, m_fullSchemaName.c_str());
            if (ECObjectsStatus::Success == schemaNameParseStatus) && m_context.IsSchemaToBePruned(key))
```

Same code but I think it's easier to read.